stdlib/public/core/IntegerParsing.swift

```diff
@@ -10,15 +10,404 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// Returns c as a UTF16.CodeUnit.  Meant to be used as _ascii16("x").
+extension FixedWidthInteger {
+  /// Creates a new integer value from the given string and radix.
+  ///
+  /// The string passed as `text` may begin with a plus or minus sign character
+  /// (`+` or `-`), followed by one or more numeric digits (`0-9`) or letters
+  /// (`a-z` or `A-Z`). Parsing of the string is case insensitive.
+  ///
+  ///     let x = Int("123")
+  ///     // x == 123
+  ///
+  ///     let y = Int("-123", radix: 8)
+  ///     // y == -83
+  ///     let y = Int("+123", radix: 8)
+  ///     // y == +83
+  ///
+  ///     let z = Int("07b", radix: 16)
+  ///     // z == 123
+  ///
+  /// If `text` is in an invalid format or contains characters that are out of
+  /// bounds for the given `radix`, or if the value it denotes in the given
+  /// `radix` is not representable, the result is `nil`. For example, the
+  /// following conversions result in `nil`:
+  ///
+  ///     Int(" 100")                       // Includes whitespace
+  ///     Int("21-50")                      // Invalid format
+  ///     Int("ff6600")                     // Characters out of bounds
+  ///     Int("zzzzzzzzzzzzz", radix: 36)   // Out of range
+  ///
+  /// - Parameters:
+  ///   - text: The ASCII representation of a number in the radix passed as
+  ///     `radix`.
+  ///   - radix: The radix, or base, to use for converting `text` to an integer
+  ///     value. `radix` must be in the range `2...36`. The default is 10.
+  @inlinable @inline(__always)
```

Comment:
How do the perf numbers compare if you make this not inlineable at all?

Generally, I'd like to see us be much less aggressive about inlining; we're getting more feedback from folks that code size is a major concern and is in fact causing serious performance problems on RAM-constrained systems.  At a minimum, could we reduce this down to a top-level switch on radix and string implementation that selects among non-inlined functions? Then any given invocation should optimize down to just a single relevant function call.

Reply:
We definitely want to be able to remove the generic type since that vast majority of call-sites will not be generic. The pattern we've often used is to have a small inlinable function that calls out to concrete implementation functions.